hystrix-microservice-provider/src/main/java/io/silverware/microservices/providers/hystrix/configuration/AnnotationScanner.java

```diff
@@ -35,76 +35,93 @@
  */
 public class AnnotationScanner {
 
+   private static final List<Class<? extends Annotation>> HIGH_LEVEL_ANNOTATIONS = Arrays.asList(Cached.class, CircuitBreaker.class, Fail.class, ThreadPool.class, Timeout.class);
+
    private AnnotationScanner() {
 
    }
 
    /**
-    * Scans given annotations and creates Hystrix service configuration.
+    * Scans given injection point for Hystrix annotations and creates Hystrix service configuration.
     *
-    * @param annotations
-    *       annotations on the injection point
+    * @param injectionPoint
+    *       injection point to be scanned
     * @return Hystrix service configuration
     */
-   public static ServiceConfig scan(Set<Annotation> annotations) {
+   public static ServiceConfig scan(InjectionPoint injectionPoint, String microserviceName) {
+      String commandKey = injectionPoint.getBean().getName() + ":" + injectionPoint.getMember().getName();
+      MethodConfig basicConfig = MethodConfig.createBuilder(microserviceName, commandKey).build();
+
+      Set<Annotation> annotations = injectionPoint.getAnnotated().getAnnotations();
       Optional<Annotation> hystrixAnnotation = annotations.stream().filter(annotation -> annotation.annotationType() == HystrixConfig.class).findAny();
+
       if (hystrixAnnotation.isPresent()) {
          HystrixConfig hystrixConfig = (HystrixConfig) hystrixAnnotation.get();
-         return scanInterfaceAnnotations(hystrixConfig.value());
+         return scanInterfaceAnnotations(hystrixConfig.value(), basicConfig);
       } else {
          Annotation[] annotationArray = new Annotation[annotations.size()];
-         return scanFieldAnnotations(annotations.toArray(annotationArray));
+         return scanFieldAnnotations(annotations.toArray(annotationArray), basicConfig);
       }
    }
 
-   private static ServiceConfig scanFieldAnnotations(Annotation[] annotations) {
-      final MethodConfig methodConfig = scanHighLevelAnnotations(annotations, null);
+   private static ServiceConfig scanFieldAnnotations(Annotation[] annotations, MethodConfig basicConfig) {
+      final MethodConfig methodConfig = scanHighLevelAnnotations(annotations, basicConfig);
       return new ServiceConfig(methodConfig);
    }
 
-   private static ServiceConfig scanInterfaceAnnotations(Class configClass) {
+   private static ServiceConfig scanInterfaceAnnotations(Class configClass, MethodConfig basicConfig) {
       final Map<String, MethodConfig> methods = new HashMap<>();
 
-      MethodConfig defaultConfig = scanClassAnnotations(configClass.getAnnotations());
+      MethodConfig defaultConfig = scanClassAnnotations(configClass.getAnnotations(), basicConfig);
 
       for (Method method : configClass.getDeclaredMethods()) {
-         MethodConfig methodConfig = scanMethodAnnotations(method, defaultConfig);
+         MethodConfig methodDefaultConfig = defaultConfig;
+         if (defaultConfig.getCommandKey().equals(basicConfig.getCommandKey())) { // TODO corner cases
```

Comment:
The problem might occur if you set `commandKey` on a class level that is the same as the automatically generated one `microserviceName:injectionPoint`. In that case it will override your explicit settings and add method name to this key. But this situation is very unlikely so I will remove the comment.